src/endpoint/scheduler.rs

```diff
@@ -202,12 +206,12 @@ impl JobHandle {
             package_name: &package.name,
             package_version: &package.version,
             log_dir: self.log_dir.as_ref(),
-            job: self.job,
+            job: self.job.clone(),
             log_receiver,
             bar: self.bar.clone(),
         }
         .join();
-        drop(self.bar);
+        drop(self.bar.clone());
```

Comment:
Yeah.
This one also gave me the chills. It is intentional. Clippy was/is kinda the one who recommended this.
The clone is to satisfy the borrow checker, but it _should_ be fine here because:

https://docs.rs/indicatif/0.17.8/indicatif/struct.ProgressBar.html
> The progress bar is an [Arc](https://doc.rust-lang.org/nightly/alloc/sync/struct.Arc.html) around its internal state. When the progress bar is cloned it just increments the refcount (so the original and its clone share the same state).

But I also don't like it, and I'm open to recommendations

Reply:
This does indeed generate a compiler error:
```
$ cargo check -r
    Checking butido v0.5.0 (/home/michael/butido-clean)
error[E0509]: cannot move out of type `JobHandle`, which implements the `Drop` trait
   --> src/endpoint/scheduler.rs:214:14
    |
214 |         drop(self.bar);
    |              ^^^^^^^^
    |              |
    |              cannot move out of here
    |              move occurs because `self.bar` has type `indicatif::ProgressBar`, which does not implement the `Copy` trait
    |
help: consider cloning the value if the performance cost is acceptable
    |
214 |         drop(self.bar.clone());
    |                      ++++++++

For more information about this error, try `rustc --explain E0509`.
error: could not compile `butido` (bin "butido") due to 1 previous error
```

I'll have to look at it later. That `clone()` definitely isn't the right solution though - you're creating an additional reference and then drop the additional reference (just a waste of time with no effect - the intention of this code was to drop the last reference).

This probably comes from the fact that `self` is mutable now but I'm surprised that this drop ever worked before (I guess the compiler is quite smart in that case).

Reply:
Ok, so this error is the result for implementing the `Drop` trait for `JobHandle` and https://doc.rust-lang.org/error_codes/E0509.html nicely explains it.

I think we can simply drop this `drop()` statement for now. It was added in 1b792a07 without an explicit explanation and I think it's just there to make the code cleaner by avoiding accidental access to the progressbar after passing it to the `LogReceiver`. The `LogReceiver` calls `self.bar.finish_with_message()` at the end of `join()` so it shouldn't cause issues if there's still a reference around (it might continue to consume some resources but it should be "locked" after that). Would be great if you could test and confirm that theory though.